src/integer_mod_q/z_q/fmpz_mod_helpers.rs

```diff
@@ -43,6 +47,113 @@ pub(crate) fn length(value: &fmpz, modulus: &fmpz) -> Z {
     }
 }
 
+unsafe impl AsInteger for Zq {
+    /// Documentation at [`AsInteger::into_fmpz`]
+    unsafe fn into_fmpz(self) -> fmpz {
+        AsInteger::into_fmpz(&self.value)
+    }
```

Comment:
Like Niklas said, it just calls the code for `Z`. Therefore, I will leave it this way.
It could be changed to
```rust
(&self.value).into_fmpz()
``` 
But I am not sure if that looks better.

Reply:
From what I understand, here you take the reference, i.e. not the efficient implementation, where you swap the value, compared to what you do for the Z value. But I am fine with this too, but this would not be the efficient variant (at least I think so :D)

Reply:
I understand what you mean now. I removed the `&` and it is now using the optimized implementation when ownership is transferred.